ray-operator/controllers/ray/rayjob_controller.go

```
@@ -449,9 +452,76 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 				return ctrl.Result{RequeueAfter: RayJobDefaultRequeueDuration}, err
 			}
 		}
+		if rayJobInstance.Spec.Schedule != "" && rayJobInstance.Status.JobDeploymentStatus != rayv1.JobDeploymentStatusFailed {
+			logger.Info("RayJob is scheduled again")
+			rayJobInstance.Status.JobDeploymentStatus = rayv1.JobDeploymentStatusScheduling
+		} else {
+			// NOTE: we do not requeue if the job fails even if scheduled, this could change
+			// If the RayJob is completed without scheduling or has failed, we should not requeue it.
+			logger.Info("RayJob is not scheduled")
+			return ctrl.Result{}, nil
+		}
+	case rayv1.JobDeploymentStatusScheduling:
+		// We attempt to delete both the cluster and the job. If they're not fully deleted,
+		// we remain in this state and continue reconciling until resources are freed.
+		// `isClusterDeleted` is initially true for when we dont need to delete our cluster
+		isClusterDeleted := true
```

isClusterDeleted defaulting to true is misleading here. I understand the intent that in the default case we don't need to delete the cluster, but it also implies the cluster is deleted which it isn't.

---

Should I change the naming? `waitForClusterDeletion`?

---

I think we can possible just remove this code? See my other comment https://github.com/ray-project/kuberay/pull/3836#discussion_r2214770286